projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6200,6 +6203,12 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
         this.summaryService.clearSummaryCache();
         this.pipeTrigger++;
         this.notifyChanges();
+        if (event.origin === TransactionEventOrigin.REDO || event.origin === TransactionEventOrigin.UNDO) {
+            event.actions.forEach(x => {
+                const value = this.transactions.getAggregatedValue(x.transaction.id, true);
+                this.validation.update(x.transaction.id, value ?? x.recordRef);
+            });
+        }
```

Comment:
Would move this above the pipe trigger and notify just for sake of consistency with the rest of the handling, though both are async but still.

Also interested if this can be combined with the selection handling that sort of checks for similar state change

Reply:
The existing selection handling seems to care only about ADD undo or DELETE redo actions, while we need to update validity on any kind of undos/redos. So I'd rather not combine them.